.ci/lib.sh

```diff
@@ -246,6 +246,26 @@ delete_stale_kata_resource()
 	done
 }
 
+delete_kata_repo_registrations()
+{
+	case "$ID" in
+		ubuntu | debian)
+			local apt_file="/etc/apt/sources.list.d/kata-containers.list"
+			if [ -f "$apt_file" ]; then
+				info "Removing Kata apt file [$apt_file]"
+				sudo rm -f "$apt_file"
+			fi
+			kata_uuid=$(sudo apt-key list | awk '$2=="home:katacontainers" {print prev} {prev=$2}')
```

Comment:
This doesn't work on my system. Should it be instead:

```sh
kata_uuid=$(sudo apt-key list | awk '$4=="home:katacontainers" {print prev} {prev=$0}')
```

There is also a warning but I'm not sure what we're supposed to do about it:

```sh
Warning: apt-key output should not be parsed (stdout is not a terminal)
```

Reply:
I suspect the warning is benign - like we have a lot of warnings about using apt in scripts as 'the interface is not guaranteed' throughout the logs iirc.
wrt $2 - better show me the output of your `sudo apt-key list`. The goal of the script was to:
- match the `home:katacontainers` to the second field of the `uid` line in the output
- and then pick up the stored second field of the *previous* line, which holds the key.

Let me post an example of what I see in a moment from my other machine..

Reply:
Here is a snippet from my 16.04 machine. Note, this snippet does not have kata in it, as my test run already nuked it from this box - but, it gives you the idea of the format we are trying to parse:

```
pub   2048R/BA07F4FB 2018-04-01 [expires: 2021-03-31]
uid                  Google Cloud Packages Automatic Signing Key <gc-team@google.com>

pub   2048R/A7317B0F 2015-04-03 [expired: 2018-04-02]
uid                  Google Cloud Packages Automatic Signing Key <gc-team@google.com>
```

Reply:
I'm running on a bionic system and that code results in `kata_uuid` being empty.

Reply:
.... so, show me the output ....

Reply:
of `sudo apt-key list` pls

Reply:
Attached using the ol' fake file extension trick ;)

![apt-key list gz](https://user-images.githubusercontent.com/20315601/53411398-407cc900-39be-11e9-847d-7d62f8d24360.png)

Reply:
right, so the format of the output on 18.04 is different from 16.04:

(from your file...)
```
pub   rsa2048 2018-03-13 [SC] [expires: 2020-08-20]
      9FDC 0CB6 3708 CF80 3696  E2DC D0B3 7B82 6063 F3ED
uid           [ unknown] home:katacontainers OBS Project <home:katacontainers@build.opensuse.org>
```

soo, I'm going to have to figure out how to parse that, and what to then give to `apt-key` to remove the key - it is reasonably different....
I'm not sure I have an 18.04 machine at my fingertips either... let me see what I can work out.

Reply:
:-( looks like it changed at 16.10. sigh, so, if apt tells you to not parse its output - but - I can't see they offer any parseable scriptable alternative either. If anybody knows (@jodh-intel - you do have history here I say ;-) ), pls let me know!

Reply:
@jodh-intel @chavafg - updated to handle both <16.10 and >16.10 apt-key output format. Can you check that for me visually and on your systems pls? I did test the awk in an 18.10 container, but it's harder for me to try the actual script. That bash string compare against "16.10" makes me twitch. Well, the whole dance makes me twitch tbh... thx!